lib/helpers/system-calls.js

```diff
@@ -0,0 +1,65 @@
+import path from 'path';
+import log from 'appium-logger';
+import support from 'appium-support';
+import { getDirectories, prettyExec, isWindows } from './helpers';
+
+class SystemCallsHelper {
+
+  constructor (sdkRoot) {
+    this.sdkRoot = sdkRoot;
+  }
+
+  async getSdkBinaryPath (binary) {
```

Comment:
this function is really long. can we break the conditional blocks out into their own helper functions?